### ISSUE TITLE
PM2_GRACEFUL_LISTEN_TIMEOUT/listen_timeout not working correctly

### ISSUE THREAD

**Issue by cdmcnamara** [Unitech/pm2#2956]
So I'm attempting to use `wait_ready` with `listen_timeout` in my app but it appears as if the `listen_timeout` is not actually working as expected. 

When attempting to see if it works as expected using the example (https://github.com/Unitech/pm2/tree/master/examples/wait-ready), it doesn't appear to be the case either.

The specific behavior that appears to be broken is even after the listen_timeout expires, pm2 still waits.

Steps to reproduce:
1) Modify the `listen_timeout` to be `1000`
2) PM2 will wait the 15 seconds in the setTimeout of app.js before displaying the status as oppose to only waiting the 1000 ms.

I've also tested this using `fork` mode as well. Is my understanding of this feature correct in that it should stop waiting and report an error status once `listen_timeout` is exceeded?

```
PM2 version: 2.5.0
Node version: 6.9.0
Mac and RHEL7
```

**Comment by Unitech** [comment]
Bug reproduced, seems like there is something missing

**Comment by Unitech** [comment]
Fixed, you can try the fix:

```bash
$ npm install Unitech/pm2#development -g
$ pm2 update
```

**Comment by cdmcnamara** [comment]
Thanks @Unitech, any idea when the next release will go out?

**Comment by Unitech** [comment]
Published on PM2 2.6.0:

```bash
$ npm install pm2@latest -g
$ pm2 update
```
